OverlayPlugin/PluginLoader.cs

```diff
@@ -62,36 +61,40 @@ public void InitPlugin(TabPage pluginScreenSpace, Label pluginStatusText)
                 return;
             }
 
+            pluginStatusText.Text = Resources.InitRuntime;
+
             Initialize();
         }
 
-        // AssemblyResolver でカスタムリゾルバを追加する前に PluginMain が解決されることを防ぐために、
-        // インライン展開を禁止したメソッドに処理を分離
+        // To prevent PluginMain from being resolved before we add our custom resolver in AssemblyResolver,
+        // we separate the reference to its own method and prevent inlining.
         [MethodImpl(MethodImplOptions.NoInlining)]
         private async void Initialize()
         {
-            pluginStatusText.Text = Resources.InitRuntime;
+            // Use TinyIoC's singleton since we don't use multiple IoC containers
```

Comment:
Unless something changed, OverlayPlugin *does* use multiple IoC containers. Every time `Initialize()` is called, a new container is created which is *very* important if you want to re-enable OverlayPlugin after disabling it.
Before this change, the second init would've started with a fresh container but after this change this would re-use the existing singleton.
I don't think there's a simple way to clear an existing container so you'd have to replace the singleton with a new instance if you want to keep the current behavior.